File: contacting-conf-web/src/main/java/nl/ing/api/contacting/conf/domain/model/surveysetting/SurveyTaskQMappingVO.java
package nl.ing.api.contacting.conf.domain.model.surveysetting;

import nl.ing.api.contacting.conf.domain.validation.SurverySettingValidation;

import java.util.Optional;

public record SurveyTaskQMappingVO(
        Optional<Long> id,
        long surveyId,
        long taskQId,
        String tqName
) {
    public SurveyTaskQMappingVO {
        SurverySettingValidation.validateId(id);
        SurverySettingValidation.validateSurveyId(surveyId);
        SurverySettingValidation.validateTaskQId(taskQId);
        SurverySettingValidation.validateTqName(tqName);
    }
}

File: contacting-conf-web/src/main/scala/nl/ing/api/contacting/conf/domain/SurveyTaskQMappingVO.scala
package nl.ing.api.contacting.conf.domain

import nl.ing.api.contacting.conf.domain.types.ContactingTypes.DatabaseId

// tq names are managed by contacting api, so string type
case class SurveyTaskQMappingVO(id: Option[DatabaseId], surveyId: DatabaseId, taskQId: DatabaseId, tqName: String)
